subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.h

```diff
@@ -145,19 +148,20 @@ static inline size_t vq_ring_size(unsigned int num, unsigned int buf_size)
 
 static inline size_t shm_size(unsigned int num, unsigned int buf_size)
 {
-	return (VDEV_STATUS_SIZE + (VRING_COUNT * vq_ring_size(num, buf_size)) +
-	       (VRING_COUNT * vring_size(num, VRING_ALIGNMENT)));
+	return (VRING_COUNT * (vq_ring_size(num, buf_size) +
+		ROUND_UP(vring_size(num, VRING_ALIGNMENT), VRING_ALIGNMENT)));
 }
 
-static inline unsigned int optimal_num_desc(size_t shm_size, unsigned int buf_size)
+static inline unsigned int optimal_num_desc(size_t mem_size, unsigned int buf_size)
 {
-	size_t available, single_alloc;
-	unsigned int num_desc;
+	size_t available;
+	unsigned int num_desc = 0;
 
-	available = shm_size - VDEV_STATUS_SIZE;
-	single_alloc = VRING_COUNT * (vq_ring_size(1, buf_size) + vring_size(1, VRING_ALIGNMENT));
+	available = mem_size - VDEV_STATUS_SIZE;
 
-	num_desc = (unsigned int) (available / single_alloc);
+	while ((available - shm_size(num_desc, buf_size)) >= shm_size(1, buf_size)) {
+		num_desc++;
+	}
```

Comment:
that's probably true without alignment, with alignment not sure.

Reply:
Uhm, why not?

`shm_size` is basically returning `VRING_COUNT * (aligned TX/RX buffers + aligned VRINGs)`

Reply:
I'm still not fully convinced by your remark but your suggestion is pretty much equivalent, so I'll implement that.